engine/src/main/java/org/terasology/world/block/items/BlockItemSystem.java

```diff
@@ -90,8 +91,16 @@ public void onPlaceBlock(ActivateEvent event, EntityRef item) {
         Vector3i placementPos = new Vector3i(targetBlock);
         placementPos.add(surfaceSide.getVector3i());
 
+        Vector3f targetPosition = event.getTargetLocation();
+        Vector2f relativePlacementPosition;
+        if (targetPosition != null) {
+            relativePlacementPosition = getSideHitPosition(event.getHitPosition(), targetPosition);
+        } else {
+            relativePlacementPosition = new Vector2f();
+        }
```

Comment:
What about moving the null check into `getSideHitPosition`?

```suggestion
        Vector2f relativePlacementPosition = getSideHitPosition(event.getHitPosition(), targetPosition);
```

Reply:
I don't like this one.
My method wants a vector to work with, nobody should pass it null.
It is true however, that I'm cluttering up the higher level method, so I moved the null check into another private method, working on the event. I think it makes more sense that way and I hope you'll like that solution too.

Reply:
I surely do :+1: